src/tests/JIT/Regression/JitBlue/Runtime_100466/Runtime_100466.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Xunit;
+
+public static class Runtime_100466
+{
+    [Fact]
+    public static int TestBoxingDoesNotTriggerStaticTypeInitializers()
+    {
+        Foo foo = new Foo();
+        ((object)foo).ToString();
+        return s_cctorTriggered ? -1 : 100;
+    }
+
+    [Fact]
+    public static int TestNullableBoxingDoesNotTriggerStaticTypeInitializers()
+    {
+        FooNullable? nullable = new FooNullable();
+        ((object)nullable).ToString();
+        return s_cctorTriggeredNullable ? -1 : 100;
+    }
+
+    private static bool s_cctorTriggered;
+    private static bool s_cctorTriggeredNullable;
+
+    private struct Foo
+    {
+        static Foo() => s_cctorTriggered = true;
+    }
+
+    private struct FooNullable
```

Comment:
Are `ref struct` types unique enough to warrant an added case here?

Reply:
The two cases in the PR provide full code coverage for the two interesting cases in the current implementation. The path taken by `ref struct`s wrt cctors is covered by the non-ref struct test cases.

It is hard to predict future interesting cases that may be introduced. If somebody changes the implementation to introduce more interesting cases in future, I hope that will do due diligence and add test coverage for it.